src/components/PostsList.vue

```diff
@@ -0,0 +1,165 @@
+<script setup>
+import { ref, onMounted } from 'vue'
+import { getPostsByUserId, deletePost, addPost, updatePost } from '@/api/posts'
+import PostsItem from './PostsItem.vue'
+import Loader from './Loader.vue'
+import SideBar from './SideBar.vue'
+
+const { userId } = defineProps({
+  userId: Number
+})
+
+const posts = ref([])
+const selectedPost = ref({})
+const editedPost = ref({})
+const isLoading = ref(false)
+const isError = ref(false)
```

Comment:
It's easier to manage a single state variable. So, instead of using separate variables like isLoading, isError, etc., I often create one variable called state, which holds a string such as 'loading', 'error', or 'success'.